Backend/components/graph_api.py

```diff
@@ -1,14 +1,49 @@
 from core import db
 from fastapi import APIRouter
+import numpy as np
+import config
 
 router = APIRouter()
 
 @router.get("/graph")
 async def get_graph(data: str, start_time: int, end_time: int):
     """pass in a comma separated list of data that the frontend requests start_time end_time must be in unix millisec"""
+    if not data:
+        # use a query to get timestamp
+        query_result = await db.query(['speed'], start_time, end_time, ['avg'],
+                                      (end_time - start_time) // 60 or 1)
+        return {'response': {"timestamps": query_result.index.to_list()}}
     data = [d.strip() for d in data.split(',')]
-    query_result = await db.query(data, start_time, end_time, ['avg']*len(data), (end_time-start_time)//5000 or 1)
-    result = {"timestamp": query_result.index.to_list()}
+    # sort the data into smaller group by aggregation methods
+    group = {'avg': [], 'first':[], 'last':[], 'max': [], 'min':[], 'no-aggr':[], 'sum': []}
     for d in data:
-        result[d] = query_result[d].to_list()
-    return result
+        # TODO change this to actual aggregation
+        group['no-aggr'].append(d) if config.FORMAT[d]['type']=='bool' else group['avg'].append(d)
```

Comment:
@MCLiii, remove or address TODO

Reply:
This will need to wait until signal spreadsheet is updated